### PR TITLE
PD-2451 Add prop types to Combobox

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
## ✍️ Proposed changes

Adds PropTypes to Combobox.

Also, adds a quick script for generating propTypes from tsdoc

🎟 _Jira ticket:_ [PD-2451](https://jira.mongodb.org/browse/PD-2451)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Style change (purely visual updates; if this is the case, attach a screenshot below!)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

## Feedback Levels

Please try to mark your comments with one of these levels

1. Praise (👏 🚀 🔥 🍾)
2. Opinions, Ideas, Suggestions (💡 👀)
3. Let’s discuss @ Code review (🤷 🤔 🧐 )
4. Doesn’t meet agreed upon standards (😬 ⚠️ ‼️)
5. Missing an edge case (🛑 🙅 ❌)
6. Missing core functionality (⛔️ ☣️ 🚧 🌋 💀)

(Note: Use ⌘-⌃-␣ to open the Emoji keyboard!)

### CODE REVIEW

File: packages/combobox/src/Combobox.tsx [hunk 1]
```diff
@@ -1351,6 +1356,51 @@ export default function Combobox<M extends boolean>({
...
+  size: PropTypes.oneOf(Object.values(ComboboxSize)),
+  disabled: PropTypes.bool,
+  description: PropTypes.string,
+  children: PropTypes.node,
+  placeholder: PropTypes.string,
+  filteredOptions: PropTypes.arrayOf(PropTypes.string),
+  // Popover Props
+  popoverZIndex: PropTypes.number,
+  usePortal: PropTypes.bool,
+  scrollContainer: PropTypes.any,
```
Comment (shaneeza) [inline packages/combobox/src/Combobox.tsx:1399]: Should we use `PropTypes.elementType`? Its what i remember you using in (LeafyGreenContext)[https://github.com/mongodb/leafygreen-ui/blob/main/packages/leafygreen-provider/src/LeafyGreenContext.tsx#L43]

**Review by shaneeza — APPROVED** [review]
Looks good! Was this generated using the script? 

Also, left one minor comment.

File: packages/combobox/src/Combobox.tsx (hunk 1 above)
Comment (TheSonOfThomp) [reply packages/combobox/src/Combobox.tsx:1399]: Yeah that makes more sense 👍 I'll make that update

**Comment by TheSonOfThomp** [comment]
> Looks good! Was this generated using the script? 
> 

The script gets us close, but there was still a bit of manual work to do

**Pull request merged**